### ISSUE TITLE
False positive on variable used before defined

### ISSUE THREAD

**Issue by srgraham** [za-creature/coffeescope#13]
This code incorrectly triggers this error:
`⚡ #8: Variable "some_variable" used before it was first defined (on line 11, column 1).`

``` coffeescript
doSomeStuff = (cb)->
  cb()
  return

if 8
  doSomeStuff ()->
    some_variable = 'eight'
    console.log some_variable
    return

some_variable = 'not eight'
console.log some_variable
```

This is as simplistic as I could break it down when I saw it triggered in one of my projects.

coffeelint.json contains this:

``` json
...
"check_scope" : {
    "module":"coffeescope2",
    "level": "warn",
    "environments": ["builtin","es5","browser","es6","node","mocha","shelljs","jquery"],
    "globals": {
        "jQuery": false,
        "_":false,
        "$":false,
        "moment":false,
        "Backbone":false,
        "Handlebars":false,
        "BLUE":true
    },
    "overwrite": false,
    "shadow": false,
    "shadow_builtins": false,
    "shadow_exceptions": ["err", "next"],
    "undefined": true,
    "unused_variables": true,
    "unused_arguments": false
  },
...
```

**Comment by za-creature** [comment]
This is mostly a documentation issue, so please leave it open until I either update the docs, or the default behavior (I'm inclined to do the later)

To fix this, you'll want to enable the [`hoist_parent`](https://github.com/za-creature/coffeescope/blob/master/src/index.coffee#L160) option which was _supposed_ to be enabled by default, but I kinda forgot about that when I made the last release :) (ref #8)

**Comment by za-creature** [comment]
New defaults (and documentation changes) landed in 0.3.0. If you're still having trouble with this, please reopen it
